src/app/public/config.ts

```diff
@@ -64,7 +64,39 @@ export interface SkyuxConfigApp {
   title?: string;
 }
 
+export interface SkyuxConfigHostFrameOptions {
+  none?: boolean;
+  self?: boolean;
+  blackbaud?: boolean;
+  urls?: string[];
+}
+
+export interface SkyuxConfigHostBBCheckout {
+  version: '2';
+}
+
+export interface SkyuxConfigHostFrameOptionsNone {
+  none: true;
+}
+
+export interface SkyuxConfigHostFrameOptionsSelf {
+  self: true;
+}
+
+export interface SkyuxConfigHostFrameOptionsBlackbaud {
+  blackbaud: true;
+}
+
+export interface SkyuxConfigHostFrameOptionsUrls {
+  urls: string[];
+}
+
 export interface SkyuxConfigHost {
+  bbCheckout?: SkyuxConfigHostBBCheckout;
+  frameOptions?: SkyuxConfigHostFrameOptionsNone |
```

Comment:
Not sure I understand the `frameOptions` type. It can be one of none, self, blackbaud, or urls? Where does `SkyuxConfigHostFrameOptions` come into play?

Reply:
Good catch.  Stand by for an update.

Reply:
I was trying to validate against properties but was running in to trouble figuring out the right TS combination to enforce that.  Enforcing it by value was successful though.